New SQLite 3.43.2.0 fails when logging exceptions if the logger framework is SLF4J 1.7

**Describe the bug**
I'm using SLF4J 1.7 and updated to the latest SQLite 3.43.2.0. When opening a connection, a `java.lang.NoSuchMethodError` exception is produced at `org.slf4j.Logger.atTrace()`

The possible reason is that SQLite 3.43.2.0 (21c77a481f6970622b99d712d36d0336d63fd554) uses the new SLF4J 2.0 fluent api (e.g. `org.slf4j.Logger.atTrace()`) that is not available in SLF4J 1.7

**To Reproduce**
Open a connection using SQLite 3.43.2.0 using SLF4J 1.7 as your logging framework: the exception is raised

**Expected behavior**
No exception should be raised

**Logs**
Log trace:
```
'org.slf4j.spi.LoggingEventBuilder org.slf4j.Logger.atTrace()'
java.lang.NoSuchMethodError: 'org.slf4j.spi.LoggingEventBuilder org.slf4j.Logger.atTrace()'
at org.sqlite.core.NativeDB.prepare(NativeDB.java:129)
at org.sqlite.core.DB.prepare(DB.java:264)
at org.sqlite.jdbc3.JDBC3Statement.lambda$execute$0(JDBC3Statement.java:53)
at org.sqlite.jdbc3.JDBC3Statement.withConnectionTimeout(JDBC3Statement.java:459)
at org.sqlite.jdbc3.JDBC3Statement.execute(JDBC3Statement.java:42)
at org.sqlite.SQLiteConfig.apply(SQLiteConfig.java:219)
at org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:70)
at org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
at org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
at org.sqlite.JDBC.createConnection(JDBC.java:106)
at org.sqlite.JDBC.connect(JDBC.java:79)
at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
```

**Environment (please complete the following information):**
 - OS: windows 11 and ubuntu in Github Actions
 - CPU architecture: x86_64
 - sqlite-jdbc version 3.43.2.0
 - slf4j-api version 1.7.36

I observe similar using `maven` + `flyway-maven-plugin`:
```
Error:  Failed to execute goal org.flywaydb:flyway-maven-plugin:9.22.3:migrate (migrate-db-before-integration-tests) on project stacey: Execution migrate-db-before-integration-tests of goal org.flywaydb:flyway-maven-plugin:9.22.3:migrate failed: An API incompatibility was encountered while executing org.flywaydb:flyway-maven-plugin:9.22.3:migrate: java.lang.NoSuchMethodError: 'org.slf4j.spi.LoggingEventBuilder org.slf4j.Logger.atTrace()'
Error:  -----------------------------------------------------
Error:  realm =    plugin>org.flywaydb:flyway-maven-plugin:9.22.3
Error:  strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
Error:  urls[0] = file:/home/runner/.m2/repository/org/flywaydb/flyway-maven-plugin/9.22.3/flyway-maven-plugin-9.22.3.jar
Error:  urls[1] = file:/home/runner/.m2/repository/org/flywaydb/flyway-core/9.22.3/flyway-core-9.22.3.jar
Error:  urls[2] = file:/home/runner/.m2/repository/com/google/code/gson/gson/2.10.1/gson-2.10.1.jar
Error:  urls[3] = file:/home/runner/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-toml/2.15.2/jackson-dataformat-toml-2.15.2.jar
Error:  urls[4] = file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.15.2/jackson-databind-2.15.2.jar
Error:  urls[5] = file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.15.2/jackson-annotations-2.15.2.jar
Error:  urls[6] = file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.15.2/jackson-core-2.15.2.jar
Error:  urls[7] = file:/home/runner/.m2/repository/org/apache/commons/commons-text/1.10.0/commons-text-1.10.0.jar
Error:  urls[8] = file:/home/runner/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar
Error:  urls[9] = file:/home/runner/work/stacey/stacey/target/classes/
Error:  urls[10] = file:/home/runner/.m2/repository/org/slf4j/slf4j-api/2.0.9/slf4j-api-2.0.9.jar
Error:  urls[11] = file:/home/runner/.m2/repository/org/xerial/sqlite-jdbc/3.43.2.0/sqlite-jdbc-3.43.2.0.jar
Error:  Number of foreign imports: 1
Error:  import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
```
with
```
Caused by: java.lang.NoSuchMethodError: 'org.slf4j.spi.LoggingEventBuilder org.slf4j.Logger.atTrace()'
    at org.sqlite.core.NativeDB.prepare (NativeDB.java:129)
    at org.sqlite.core.DB.prepare (DB.java:264)
    at org.sqlite.jdbc3.JDBC3Statement.lambda$execute$0 (JDBC3Statement.java:53)
    at org.sqlite.jdbc3.JDBC3Statement.withConnectionTimeout (JDBC3Statement.java:459)
    at org.sqlite.jdbc3.JDBC3Statement.execute (JDBC3Statement.java:42)
    at org.sqlite.SQLiteConfig.apply (SQLiteConfig.java:219)
    at org.sqlite.SQLiteConnection.<init> (SQLiteConnection.java:70)
    at org.sqlite.jdbc3.JDBC3Connection.<init> (JDBC3Connection.java:28)
    at org.sqlite.jdbc4.JDBC4Connection.<init> (JDBC4Connection.java:19)
    at org.sqlite.JDBC.createConnection (JDBC.java:106)
    at org.sqlite.JDBC.connect (JDBC.java:79)
```
Just guessing - classworlds lists slf4j-api v2 but I suppose it's too late as maven itself uses slf4j v1.7 and `Logger` is (probably) already loaded and configured at this moment.

--- a/Makefile
+++ b/Makefile
@@ -45,7 +45,7 @@ $(SQLITE_UNPACKED): $(SQLITE_ARCHIVE)
 
 $(JAVA_CLASSPATH):
 	@mkdir -p $(@D)
-	curl -L -f -o$@ https://search.maven.org/remotecontent?filepath=org/slf4j/slf4j-api/2.0.9/slf4j-api-2.0.9.jar
+	curl -L -f -o$@ https://search.maven.org/remotecontent?filepath=org/slf4j/slf4j-api/1.7.36/slf4j-api-1.7.36.jar
 
 $(TARGET)/common-lib/org/sqlite/%.class: src/main/java/org/sqlite/%.java
 	@mkdir -p $(@D)

--- a/Makefile.common
+++ b/Makefile.common
@@ -18,7 +18,7 @@ JAVA_CLASSPATH:=$(TARGET)/classpath/slf4j-api.jar
 ifeq ("$(wildcard $(OSINFO_PROG))","")
 $(info Building OSInfo tool)
 $(shell mkdir -p $(TARGET)/classpath)
-$(shell curl -L -f -o$(JAVA_CLASSPATH) https://search.maven.org/remotecontent?filepath=org/slf4j/slf4j-api/2.0.9/slf4j-api-2.0.9.jar)
+$(shell curl -L -f -o$(JAVA_CLASSPATH) https://search.maven.org/remotecontent?filepath=org/slf4j/slf4j-api/1.7.36/slf4j-api-1.7.36.jar)
 $(shell $(JAVAC) -cp $(JAVA_CLASSPATH) -sourcepath $(SRC) -d lib src/main/java/org/sqlite/util/OSInfo.java)
 endif
 

--- a/pom.xml
+++ b/pom.xml
@@ -408,7 +408,7 @@
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
-            <version>2.0.9</version>
+            <version>1.7.36</version>
         </dependency>
         <!--
         This dependency makes compilation on non-GraalVM versions possible.

--- a/src/main/java/org/sqlite/JDBC.java
+++ b/src/main/java/org/sqlite/JDBC.java
@@ -30,7 +30,7 @@ public class JDBC implements Driver {
         try {
             DriverManager.registerDriver(new JDBC());
         } catch (SQLException e) {
-            logger.atError().setCause(e).log();
+            logger.error("Could not register driver", e);
         }
     }
 

--- a/src/main/java/org/sqlite/SQLiteJDBCLoader.java
+++ b/src/main/java/org/sqlite/SQLiteJDBCLoader.java
@@ -101,14 +101,12 @@ public class SQLiteJDBCLoader {
                                     try {
                                         Files.delete(nativeLib);
                                     } catch (Exception e) {
-                                        logger.atError()
-                                                .setCause(e)
-                                                .log("Failed to delete old native lib");
+                                        logger.error("Failed to delete old native lib", e);
                                     }
                                 }
                             });
         } catch (IOException e) {
-            logger.atError().setCause(e).log("Failed to open directory");
+            logger.error("Failed to open directory", e);
         }
     }
 
@@ -224,7 +222,7 @@ public class SQLiteJDBCLoader {
             }
             return loadNativeLibrary(targetFolder, extractedLibFileName);
         } catch (IOException e) {
-            logger.atError().setCause(e).log();
+            logger.error("Unexpected IOException", e);
             return false;
         }
     }
@@ -248,7 +246,7 @@ public class SQLiteJDBCLoader {
             connection.setUseCaches(false);
             return connection.getInputStream();
         } catch (IOException e) {
-            logger.atError().setCause(e).log();
+            logger.error("Could not connect", e);
             return null;
         }
     }
@@ -268,12 +266,12 @@ public class SQLiteJDBCLoader {
                 System.load(new File(path, name).getAbsolutePath());
                 return true;
             } catch (UnsatisfiedLinkError e) {
-                logger.atError()
-                        .setCause(e)
-                        .setMessage("Failed to load native library: {}. osinfo: {}")
-                        .addArgument(name)
-                        .addArgument(OSInfo::getNativeLibFolderPathForCurrentOS)
-                        .log();
+
+                logger.error(
+                        "Failed to load native library: {}. osinfo: {}",
+                        name,
+                        OSInfo.getNativeLibFolderPathForCurrentOS(),
+                        e);
                 return false;
             }
 
@@ -287,9 +285,7 @@ public class SQLiteJDBCLoader {
             System.loadLibrary(LibraryLoaderUtil.NATIVE_LIB_BASE_NAME);
             return true;
         } catch (UnsatisfiedLinkError e) {
-            logger.atError()
-                    .setCause(e)
-                    .log("Failed to load native library through System.loadLibrary");
+            logger.error("Failed to load native library through System.loadLibrary", e);
             return false;
         }
     }
@@ -422,9 +418,7 @@ public class SQLiteJDBCLoader {
                 // inline creation of logger to avoid build-time initialization of the logging
                 // framework in native-image
                 LoggerFactory.getLogger(VersionHolder.class)
-                        .atError()
-                        .setCause(e)
-                        .log("Could not read version from file: {}", versionFile);
+                        .error("Could not read version from file: {}", versionFile, e);
             }
             VERSION = version;
         }

--- a/src/main/java/org/sqlite/core/NativeDB.java
+++ b/src/main/java/org/sqlite/core/NativeDB.java
@@ -90,11 +90,10 @@ public final class NativeDB extends DB {
     /** @see org.sqlite.core.DB#_exec(java.lang.String) */
     @Override
     public synchronized int _exec(String sql) throws SQLException {
-        logger.atTrace()
-                .setMessage("DriverManager [{}] [SQLite EXEC] {}")
-                .addArgument(() -> Thread.currentThread().getName())
-                .addArgument(sql)
-                .log();
+        if (logger.isTraceEnabled()) {
+            logger.trace(
+                    "DriverManager [{}] [SQLite EXEC] {}", Thread.currentThread().getName(), sql);
+        }
         return _exec_utf8(stringToUtf8ByteArray(sql));
     }
 
@@ -126,11 +125,10 @@ public final class NativeDB extends DB {
     /** @see org.sqlite.core.DB#prepare(java.lang.String) */
     @Override
     protected synchronized SafeStmtPtr prepare(String sql) throws SQLException {
-        logger.atTrace()
-                .setMessage("DriverManager [{}] [SQLite EXEC] {}")
-                .addArgument(() -> Thread.currentThread().getName())
-                .addArgument(sql)
-                .log();
+        if (logger.isTraceEnabled()) {
+            logger.trace(
+                    "DriverManager [{}] [SQLite EXEC] {}", Thread.currentThread().getName(), sql);
+        }
         return new SafeStmtPtr(this, prepare_utf8(stringToUtf8ByteArray(sql)));
     }
 

--- a/src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java
+++ b/src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java
@@ -964,14 +964,14 @@ public abstract class JDBC3DatabaseMetaData extends CoreDatabaseMetaData {
                         try {
                             rsColAutoinc.close();
                         } catch (Exception e) {
-                            LogHolder.logger.atError().setCause(e).log();
+                            LogHolder.logger.error("Could not close ResultSet", e);
                         }
                     }
                     if (statColAutoinc != null) {
                         try {
                             statColAutoinc.close();
                         } catch (Exception e) {
-                            LogHolder.logger.atError().setCause(e).log();
+                            LogHolder.logger.error("Could not close statement", e);
                         }
                     }
                 }
@@ -1122,7 +1122,7 @@ public abstract class JDBC3DatabaseMetaData extends CoreDatabaseMetaData {
                 try {
                     rs.close();
                 } catch (Exception e) {
-                    LogHolder.logger.atError().setCause(e).log();
+                    LogHolder.logger.error("Could not close ResultSet", e);
                 }
             }
         }

--- a/src/main/java/org/sqlite/jdbc3/JDBC3Statement.java
+++ b/src/main/java/org/sqlite/jdbc3/JDBC3Statement.java
@@ -97,7 +97,7 @@ public abstract class JDBC3Statement extends CoreStatement {
         private static final Logger logger = LoggerFactory.getLogger(BackupObserver.class);
 
         public void progress(int remaining, int pageCount) {
-            logger.atInfo().log("remaining:{}, page count:{}", remaining, pageCount);
+            logger.info("remaining:{}, page count:{}", remaining, pageCount);
         }
     }
 

--- a/src/main/java/org/sqlite/util/OSInfo.java
+++ b/src/main/java/org/sqlite/util/OSInfo.java
@@ -160,7 +160,7 @@ public class OSInfo {
         try {
             return processRunner.runAndWaitFor("uname -m");
         } catch (Throwable e) {
-            LogHolder.logger.atError().setCause(e).log("Error while running uname -m");
+            LogHolder.logger.error("Error while running uname -m", e);
             return "unknown";
         }
     }
@@ -221,10 +221,8 @@ public class OSInfo {
                         return "armv7";
                     }
                 } else {
-                    LogHolder.logger
-                            .atWarn()
-                            .log(
-                                    "readelf not found. Cannot check if running on an armhf system, armel architecture will be presumed");
+                    LogHolder.logger.warn(
+                            "readelf not found. Cannot check if running on an armhf system, armel architecture will be presumed");
                 }
             } catch (IOException | InterruptedException e) {
                 // ignored: fall back to "arm" arch (soft-float ABI)

